docs/src/building-rust-gpu.md

```diff
@@ -19,6 +19,13 @@
 
     This will build `rustc_codegen_spirv`, the compiler, then use that compiler to build [`sky-shader`](examples/shaders/sky-shader) into a SPIR-V module, then finally, build a `wgpu` sample app (modified from [`wgpu`'s examples](https://github.com/gfx-rs/wgpu-rs/tree/master/examples/hello-triangle)) using the built SPIR-V module to display the shader in a window.
 
+## Prerequisite linux packages recommended to install before building Rust-GPU
+
+You may need the development version (i.e. headers/etc. included) of some packages in some distributions to be able to
+build the examples - specifically, x11 and libxkbcommon, as well as gcc/clang with c++ support. These packages may be
+called (fedora) `libX11-devel`, `libxkbcommon-x11-devel`, and `gcc-c++`, or (ubuntu) `libxkbcommon-x11-dev`,
+`libx11-dev`, and `gcc`.
```

Comment:
Looks like those correspond to these: https://github.com/EmbarkStudios/rust-gpu/blob/56ba2d2afa5a64db26cac631d33c42fca521f1d2/default.nix#L12-L14

There's also these for running e.g. `example-runner-wgpu`: https://github.com/EmbarkStudios/rust-gpu/blob/56ba2d2afa5a64db26cac631d33c42fca521f1d2/default.nix#L16-L19

However, not including them is probably fine for traditional Linux distributions, because a graphical environment will likely have all of those available already, so I don't think we have to say anything about it.

(As a random aside, the `default.nix` file was tested with `nix-shell --pure`, which hides anything available in the ambient environment - so it should be complete, tho I should probably retest every time we update deps)